### ISSUE TITLE
Conan with CMake, always uses system-wide Boost

### ISSUE THREAD

**Issue by SamuelMarks** [conan-io/conan#304]
```
cmake_minimum_required(VERSION 3.5)
project(hello_conan)

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")

if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/build/conanbuildinfo.cmake)
    include(${CMAKE_CURRENT_SOURCE_DIR}/build/conanbuildinfo.cmake)
else()
    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake) # Not CLion
endif()

find_package(Boost 1.60.0 REQUIRED)

set(SOURCE_FILES main.cpp)
add_executable(hello_conan ${SOURCE_FILES})
```

```
[requires]
Boost/1.60.0@lasote/stable

[generators]
cmake
```

```
#include <iostream>
#include <boost/version.hpp>

int main(void) {
    std::cout << "Using Boost " << BOOST_VERSION / 100000 << "."
                   << BOOST_VERSION / 100 % 1000 << "."
                   << BOOST_VERSION % 100 << std::endl;
}
```

```
hello_conan $ mkdir build && cd $_
hello_conan/build $ conan install ..
```

But running in CLion gives me an error. Tying with `cmake .. -G 'Unix Makefiles'` gives same error:
```
-- The C compiler identification is GNU 5.3.1
-- The CXX compiler identification is GNU 5.3.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/share/cmake-3.5/Modules/FindBoost.cmake:1677 (message):
  Unable to find the requested Boost libraries.

  Boost version: 1.58.0

  Boost include path: /usr/include

  Detected version of Boost is too old.  Requested version was 1.60 (or
  newer).
Call Stack (most recent call first):
  CMakeLists.txt:13 (find_package)


-- Configuring incomplete, errors occurred!
```

Even tried manually include `~/.conan/data/Boost/1.60.0/lasote/stable/export/FindBoost.cmake` in my `CMakeLists.txt`, which got past the `find_package(Boost 1.60.0 REQUIRED)` constraint but the actual boost version imported was 1.58.0.

How do I automatically find the conan include directory?

**Comment by memsharded** [comment]
The fact is that the ``include(${CMAKE_CURRENT_SOURCE_DIR}/build/conanbuildinfo.cmake)`` does not do anything at all, but declare some CMake variables, more specifically some ``CONAN_XXX_`` variables. So it does not have any side effect at all related to CMake. This is by design. 

The actual job is done by the ``conan_basic_setup()`` which in your case, you are not calling before the find boost. Inside the ``conan_basic_setup()`` you will find that it is calling the ``include_directories()``, but you will find also, that it is prepending the packages folders to ``CMAKE_MODULE_PATH``. 

In this way, the packages paths are looked for first, instead of the system ones. The Boost conan package has a Find.cmake inside, so your code should work either adding the ``conan_basic_setup()`` call, or do it yourself in your CMakeLists.txt, something like:

```cmake
if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/build/conanbuildinfo.cmake)
    include(${CMAKE_CURRENT_SOURCE_DIR}/build/conanbuildinfo.cmake)
else()
    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake) # Not CLion
endif()

set(CMAKE_MODULE_PATH ${CONAN_CMAKE_MODULE_PATH} ${CMAKE_MODULE_PATH})

find_package(Boost 1.60.0 REQUIRED)
```
You might also try to use Boost without the ``find_package(Boost)``, after all, you have the include directories, the library names...

Please try this and tell me if it worked, thanks!

**Comment by memsharded** [comment]
Actually, I have just merged https://github.com/conan-io/conan/pull/292, by @osechet, that introduces a convenient (and simpler to remember) ``conan_set_find_paths()`` macro, that you could call instead of the ``set``. Already in develop, to be released in 0.10.

**Comment by SamuelMarks** [comment]
Perfect thanks
